src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.component.ts

```diff
@@ -89,13 +89,15 @@ export class LectureUnitManagementComponent implements OnInit, OnDestroy {
         });
 
         this.updateOrderSubject = new Subject();
-        this.activatedRoute.parent!.params.subscribe((params) => {
-            this.lectureId ??= +params['lectureId'];
-            if (this.lectureId) {
-                // TODO: the lecture (without units) is already available through the lecture.route.ts resolver, it's not really good that we load it twice
-                this.loadData();
-            }
-        });
+        if (this.activatedRoute?.parent?.params) {
+            this.activatedRoute.parent.params.subscribe((params) => {
+                this.lectureId ??= +params['lectureId'];
+                if (this.lectureId) {
+                    // TODO: the lecture (without units) is already available through the lecture.route.ts resolver, it's not really good that we load it twice
+                    this.loadData();
+                }
+            });
+        }
```

Comment:
You should be able to remove the if-statement when using this `this.activatedRoute?.parent?.params.subscribe((params) => { ... `

Can you unsubscribe from it again?